nowcasting_api/database_fast.py

```diff
@@ -0,0 +1,92 @@
+""" Get data from database - optimized"""
+
+from datetime import datetime
+
+from nowcasting_datamodel.models import ForecastValueLatestSQL, MLModelSQL
+from pydantic_models import OneDatetimeManyForecastValues
+from sqlalchemy.orm.session import Session
+
+
+def get_forecast_values_all_compact(
+    session: Session,
+    start_datetime_utc: datetime | None = None,
+    end_datetime_utc: datetime | None = None,
+    gsp_ids=None,
```

Comment:
Should this type be `Optional[str] = None` rather than just `None`?

Reply:
Optional[str] was the old(er) way to do it, but now it tends be done like this.

Reply:
Just saw this as I was closing tabs – surely this just isn't typing `gsp_ids` here then, instead of the optional datetime params above that are `datetime | None` (should `gsp_ids` be `gsp_ids: str | None` for better type inference?)

Reply:
yea, it should be infact `gsp_ids: [str] | None`, ill make an issue for this

Reply:
https://github.com/openclimatefix/uk-pv-national-gsp-api/issues/458